### PR TITLE
Simplify tests for org_dogs_test

### PR DESCRIPTION
**Pull request by Eduardo06sp**
**Summary of changes:**
- Added size/sex/description to `one` & `paused_application` dog fixtures so they wouldn't have to be added during tests

- When testing that verified staff can unpause a paused dog, I used the existing `paused_application` dog, which got rid of having to pause it during the test

- Created an image fixture attachment for `dog_one`, which helped simplify tests so you don't have to manually upload the images each time

- Separated tests a few times

- Removed some unnecessary params throughout the tests

- Oh, and added a missing newline to AttachmentsController :}

**Further explanation for changes:**

A notable change was setting up the dog's image attachment fixture. I decided this would be best when it came to testing the removal of photos because it got rid of the step where you upload images. There is no need to keep uploading images when we have a separate test that verifies that we can successfully upload images.

Additionally, it may start to become clear that certain tests can be extracted into new, separate files as suggested by Jason Swett. I did not want to introduce a large amount of changes for this, so I kept everything here. It can be done in a separate PR.

**Note:** I do realize that I changed the fundamental approach to a few of these tests while refactoring. So this PR also serves as kind of a proposal for their approach.

Do not hesitate to ask for more clarification, or let me know if you would like to keep certain things just like they were before!

**Related:** #149

### CODE REVIEW

File: test/integration/org_dogs_test.rb [hunk 1]
```diff
@@ -109,152 +109,116 @@ class OrgDogsTest < ActionDispatch::IntegrationTest
...
-          sex: 'Female',
-          breed: 'mix',
-          size: 'Medium (22-57 lb)',
-          description: 'A lovely little pooch this one.',
-          append_images:
-          [
-            fixture_file_upload("test.png", "image/png"),
-            fixture_file_upload("test2.png", "image/png")
-          ]
+    assert_difference '@dog.images_attachments.length', 2 do
```
Comment (kasugaijin) [inline test/integration/org_dogs_test.rb:174]: this is cleaner than how I would have done it :)

**Comment by kasugaijin** [comment]
Thanks @Eduardo06sp I like your approach here and think it's good to merge. I agree with decoupling some of the tested actions and testing them in a separate case (e.g., pausing dogs and verifying pause reason). 

I also like adding the attachment to the dog fixture to remove the need for having to do that in the test, as well as the other attributes like size, breed etc. This cleans up the tests a lot. 

An interesting note, not related to this PR, but related to Jason's code review, was the approach of decoupling entirely from the DOM and checking the response for presence of certain attributes. In a lot of cases we are looking for a certain DOM element and then asserting its value. This could be improved by ignoring the DOM entirely and checking the server response markup for the value we expect. Not something we need to do in this test suite, but something I will consider in future. Counter to that, though, I think it is convention in a lot of codebases to look for DOM elements and assert their values. So it's probably a matter of opinion with pros and cons for each.

**Comment by Eduardo06sp** [comment]
@kasugaijin I don't like the approach of relying on the server's response to check for the relevant text to display. If the test fails, the error message is really unhelpful.

You can get a lot more useful error messages by choosing a more general selector and using regex to validate the inclusion of the text string anywhere in the matched element.

E.g. 
```rb
assert_select "body", /our dogs/
```
in one of the tests generates this helpful error (where I can easily read through it):
image

compared to the error generated by
```rb
assert_includes @response.body, "our dogs"
```
image
I actually can't take a screenshot of all of it since I have to scroll down the equivalent of 3-4 screen lengths!

I personally prefer just using specific selectors (I prefer choosing a descriptive class name over something like `div.col-lg-4` when available). This generates the most useful feedback upon failure in my opinion.

However, as you can see, choosing a more general selector is still very simple & does not add much complication. It is up to you, these are just my thoughts. :}

**Comment by kasugaijin** [comment]
Yes you are totally right. I think it's worth the cost of some coupling with the DOM for the sake of easier debugging and more useful error messages. Great point!

**Pull request merged**